flax/nnx/variablelib.py

```diff
@@ -1133,6 +1138,9 @@ def _check_can_update(self):
       )
 
   def __getattr__(self, name: str) -> tp.Any:
+    if name == 'sharding_names': # for backward compatibility
+      warnings.warn("'sharding_names' is deprecated. Use 'sharding_metadata' instead.", DeprecationWarning)
+      return self.sharding_metadata
```

Comment:
we should use `has_metadata` and `get_metadata` in the lines bellow and teach these methods how to handle `sharding_names`.